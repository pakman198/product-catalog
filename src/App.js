import React from 'react';

import CatalogProvider from './CatalogProvider';
import ProductList from './ProductList';
import ProductDetail from './ProductDetail';
import CatalogContext from './CatalogContext';
import Header from './Header';

const App = () => {
    return (
      <CatalogProvider>
        <div className="ui container">
          <Header />
          <CatalogContext.Consumer>
            {
              ({selectedProduct}) => {
                return selectedProduct ? <ProductDetail /> : null
              }
            }
          </CatalogContext.Consumer>
          <ProductList />
        </div>
      </CatalogProvider>
    );
}

export default App;
