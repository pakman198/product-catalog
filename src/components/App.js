import React from 'react';

import ProductList from './ProductList';
import Header from './Header';
import ModalProvider from './ModalProvider';

const App = () => {
    return (
        <div className="ui container">
          <Header />
          <ModalProvider>
            <ProductList />
          </ModalProvider>
        </div>
    );
}

export default App;
