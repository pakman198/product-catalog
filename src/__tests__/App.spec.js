import React from 'react'
import { render } from '@testing-library/react'

import App from '../components/App';

test('App.js', () => {
  const { asFragment } = render(<App />);
  
  expect(asFragment()).toMatchSnapshot();
});