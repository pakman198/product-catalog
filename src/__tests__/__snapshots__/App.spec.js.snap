// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js 1`] = `
<DocumentFragment>
  <div
    class="ui container"
  >
    <div
      class="ui borderless top fixed menu"
    >
      <div
        class="item"
      >
        <i
          class="large icon fly"
        />
        WestElm
      </div>
    </div>
    <div
      class="product-list ui stackable three column grid"
    />
  </div>
</DocumentFragment>
`;
